dvc/repo/reproduce.py

```diff
@@ -139,12 +127,18 @@ def reproduce(  # noqa: C901
             )
 
     result = _reproduce_stages(self.index.graph, list(stages), **kwargs)
-    _track_top_level(self)
```

Comment:
Do you think it is worth adding a general test that ensures things are not staged during `dvc repro`?

Reply:
I don't think so, because unless `core.autostage` is set, repro should not autostage.  In the same way that `dvc add`, etc should not. It's unexpected.

Reply:
Yes, I meant if it was worth adding a check as a way to catch that we don't accidentally break this rule, like the case we are fixing here.
If you don't really expect it to happen again, I am ok, but feels like maybe a simple check in one of the existing test would serve as safeguard